cookbooks/cdo-varnish/templates/default/default.vcl.erb

```diff
@@ -77,42 +77,56 @@ sub vcl_recv
   }
 ###
 
-  # Normalize Accept-Language header into X-Varnish-Accept-Language for better cache hit rates.
-  # To vary content based on the language, set Vary: X-Varnish-Accept-Language in application code.
-  C{
-      vcl_rewrite_accept_language(ctx);
-  }C
-
-### EXTRACT COOKIE HEADERS
+### REQUEST COOKIE+HEADER FILTER
 if(req.http.Cookie) {
   cookie.parse(req.http.Cookie);
-<% node['cdo-varnish']['cookie_headers'].to_hash.each do |cookie, header| %>
-  if(cookie.isset("<%= cookie %>")) {
-    set req.http.<%= header %> = cookie.get("<%= cookie %>");
-    cookie.delete("<%= cookie %>");
+}
+<%= config = node['cdo-varnish']['config']
+    setup_behavior config, 'request' %>
+if(req.http.Cookie) {
+  set req.http.Cookie = cookie.get_string();
+  if (req.http.Cookie == "") {
+    unset req.http.Cookie;
   }
-<%end%>
 }
 ###
 
+# Normalize Accept-Language header into X-Varnish-Accept-Language for better cache hit rates.
+C{
+  vcl_rewrite_accept_language(ctx);
+}C
+
+# Collapse cookie-extracted language setting into X-Varnish-Accept-Language header if present.
+if (req.http.X-Cookie-language_) {
+  set req.http.X-Varnish-Accept-Language = req.http.X-Cookie-language_;
+  unset req.http.X-Cookie-language_;
+}
+
 ### PROXIES
-<%= config = node['cdo-varnish']['config']
-    setup_behavior config, 'proxy' %>
+<%= setup_behavior config, 'proxy' %>
 ###
 
-### REQUEST COOKIE FILTER
-if (req.http.Cookie) {
-  # Only apply cookie filter to cached HTTP methods.
-  if(req.method == "GET" || req.method == "HEAD") {
-  <%= setup_behavior config, 'request' %>
-    set req.http.Cookie = cookie.get_string();
-    if (req.http.Cookie == "") {
-      unset req.http.Cookie;
-    }
-  }
+# Override builtin to allow caching whitelisted cookies.
+if (req.method != "GET" &&
+  req.method != "HEAD" &&
+  req.method != "PUT" &&
+  req.method != "POST" &&
+  req.method != "TRACE" &&
+  req.method != "OPTIONS" &&
+  req.method != "DELETE") {
+  /* Non-RFC2616 or CONNECT which is weird. */
+  return (pipe);
```

Comment:
Yes, makes sense to change the behavior to match CloudFront here, I'll change this in the next commit.